api/src/main/java/com/datastrato/gravitino/rel/Table.java

```diff
@@ -58,4 +58,15 @@ default String comment() {
   default Map<String, String> properties() {
     return Collections.emptyMap();
   }
+
+  /**
+   * Table method for working with partitions. If the table does not support partition operations,
+   * an {@link UnsupportedOperationException} is thrown.
+   *
+   * @return The partition support table.
+   * @throws UnsupportedOperationException If the table does not support partition operations.
+   */
+  default SupportsPartitions supportPartitions() throws UnsupportedOperationException {
```

Comment:
Is better to rename the method name to `asPartitioned`, then the client can use like `table.asPartitioned().xxxPartition(xx, xx)`